dashboard/src/modules/components/DatePickerComponent/index.jsx

```diff
@@ -48,8 +53,11 @@ function DatePickerWidget({
       />
       <InputGroupText>to</InputGroupText>
       <DatePicker
-        value={formatDate(toDate)}
-        onChange={(_str,date) => setToDate(date)}
+        value={strDate}
+        onChange={(_str) => {
+          setStrDate(_str);
+          setToDate(constructUTCDate(new Date(_str)));
+        }}
```

Comment:
I would be disappointed if LGTM flagged an unused parameter in this context, since you don't control the signature of the callback, you're only supplying an implementation of it.

However, your branch currently has merge conflicts, so LGTM is refusing to analyze it.  Could you update your local repo with the current `main` branch (i.e, `git remote update --prune`) and then rebase your branch on the end of it (and then force-push the result)?  Then we'll know....  🙂